test/github-release.ts

```diff
@@ -41,60 +41,86 @@ describe('GitHubRelease', () => {
   afterEach(() => {
     sandbox.restore();
   });
-  const stubDefaultBranch = (github: GitHub) => {
-    sandbox.stub(github, 'getDefaultBranch').resolves('main');
-  };
-
-  const stubPRs = (github: GitHub, head: string, title: string) => {
-    sandbox.stub(github, 'findMergedPullRequests').resolves([
-      {
-        sha: 'abc123',
-        number: 1,
-        baseRefName: 'main',
-        headRefName: head,
-        labels: ['autorelease: pending'],
-        title: title,
-        body: 'Some release notes',
-      },
-    ]);
-  };
-
-  const stubChangeLog = (
-    github: GitHub,
-    file = 'CHANGELOG.md',
-    version = '1.0.3'
-  ): sinon.SinonStub => {
-    const stub = sandbox.stub(github, 'getFileContentsOnBranch');
-    stub
-      .withArgs(file, 'main')
-      .resolves(buildFileContent(`#Changelog\n\n## v${version}\n\n* entry`));
-    return stub;
-  };
-
-  const stubCommentsAndLabels = (github: GitHub) => {
-    sandbox
-      .stub(github, 'commentOnIssue')
-      .withArgs(':robot: Release is at https://release.url :sunflower:', 1)
-      .resolves();
-
-    sandbox
-      .stub(github, 'addLabels')
-      .withArgs(['autorelease: tagged'], 1)
-      .resolves();
-
-    sandbox.stub(github, 'removeLabels').withArgs(['autorelease: pending'], 1);
-  };
-
-  describe('createRelease', () => {
+  function mockGithub(options: {
```

Comment:
I'd be tempted to give this a slightly more unwieldy name that includes the specific functionality being mocked on GitHub, doesn't have to be everything:

```
mockGitHubIssueUpdates?
```

Reply:
Do you mean splitting into something like

`mockGitHubCommon` that mocks the stuff every test that calls it needs mocked: `getRepositoryDefaultBranch`, `findMergedPullRequests`, `getFileContentsOnBranch` and separately one that isn't called by every test:
`mockGitHubCommentAndLabels`: that mocks `commentOnIssue`, `addLabels`, `removeLabels`

Or do you think there should be more granular, perhaps per method, mock functions? (e.g. `mockGitHubDefaultBranch()`, `mockGitHubPRs()`, etc). I could see it either way really.

Reply:
fyi I went with No.1

Reply:
I like `mockGitHubCommon` 👍